src/sentry/api/endpoints/organization_details.py

```diff
@@ -266,7 +265,12 @@ def validate_projectRateLimit(self, value):
     def validate_providerName(self, value):
         from sentry.auth import manager
 
-        if not manager.exists(value):
+        organization = self.context["organization"]
+        request = self.context["request"]
+        has_auth_provider_config = features.has(
+            "organizations:auth-provider-config", organization, actor=request.user
+        )
+        if has_auth_provider_config and not manager.exists(value):
```

Comment:
We're combining 2 checks here (feature flag and key validation) which makes it slightly harder to read. 

It will be better to do an [early-return](https://medium.com/swlh/return-early-pattern-3d18a41bba8) for the feature flag check.

Reply:
Changed to early-return pattern:

```
    if not has_api_auth_provider:
        return
    if not manager.exists(value):
        raise serializers.ValidationError("Invalid providerName")
    return value
```